### ISSUE TITLE
retry does not work with JUnitParamsRunner

### ISSUE THREAD

**Issue by sereneliu** [gradle/test-retry-gradle-plugin#101]
I get this error message for tests with JUnitParamsRunner:

`org.gradle.test-retry was unable to retry the following test methods, which is unexpected. Please file a bug report at https://github.com/gradle/test-retry-gradle-plugin/issues`

What can I tweak to make this work?

**Comment by daknin** [comment]
I've also encountered this. It looks like the `JUnitParamsRunner` sets the test names to be something like `parameterizedMethod(1) [0]` which `BaseJunitTestFrameworkStrategy::addPotentiallyParameterizedSuffixed` incorrectly changes to  `parameterizedMethod(1) ` instead of `parameterizedMethod`

Changing `String strippedParameterName = name.replaceAll("(?:\\([^)]*?\\)|\\[[^]]*?])*$", "");` to `String strippedParameterName = name.replaceAll("(?:\\s?\\([^)]*?\\)|\\s?\\[[^]]*?])*$", "");` seems to fix it

Sample test
```java
package org.example;

import junitparams.JUnitParamsRunner;
import junitparams.Parameters;
import junitparams.naming.TestCaseName;
import org.junit.Test;
import org.junit.experimental.categories.Category;
import org.junit.runner.RunWith;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.*;

@RunWith(JUnitParamsRunner.class)
public class MainTest {
	@Test
	@Parameters({
			"17, true",
			"22, true" })
	@TestCaseName("parameterized{0}_{1}")
	public void parameterizedMethod(int value, boolean valid) {
		assertThat(value > 30, is(valid));
	}
}
```

build.gradle
```groovy
plugins {
    id 'java'
	id "org.gradle.test-retry" version "1.4.0"
}

group 'org.example'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

dependencies {
	testImplementation 'junit:junit:4.13.2'
	implementation 'pl.pragmatists:JUnitParams:1.1.1'
	testImplementation 'org.junit.vintage:junit-vintage-engine:5.9.0'

}

test {
	retry {
		maxRetries.set(2)
		maxFailures.set(20)
		failOnPassedAfterRetry.set(true)
	}
    useJUnit() {
	}
}

```

**Comment by marcphilipp** [comment]
@daknin Thanks for the analysis and the workaround! Since method names (at least in Java) normally don't contain parentheses, I can imagine we might be able to remove an optional `(.*) ` suffix. Do you have time to try out this change and see of it breaks any existing test?

**Comment by jean-andre-gauthier** [comment]
This has been fixed in https://github.com/gradle/test-retry-gradle-plugin/pull/272, and will be released in `1.5.8`
